src/java/org/jivesoftware/openfire/roster/RosterManager.java

```diff
@@ -592,7 +592,21 @@ public void userDeleting(User user, Map<String,Object> params) {
     }
 
     public void userModified(User user, Map<String,Object> params) {
-        //Do nothing
+        if ("nameModified".equals(params.get("type"))) {
+
+            for (Group group : getSharedGroups(user.getUsername())) {
+                ArrayList<JID> groupUsers = new ArrayList<>();
+                groupUsers.addAll(group.getAdmins());
+                groupUsers.addAll(group.getMembers());
+
+                for (JID groupUser : groupUsers) {
+                    String groupUsername = groupUser.getNode();
+                    synchronized ((groupUsername + " ro").intern()) {
```

Comment:
I don't think this is required. Cache operations are thread-safe (as per JavaDoc).